app/controllers/hyrax/dashboard/collections_controller.rb

```diff
@@ -224,12 +208,39 @@ def create_valkyrie_collection
         after_create_response
       end
 
-      def valkyrie_update
-        form.validate(collection_params) &&
-          @collection = transactions['change_set.update_collection']
-                        .call(form)
-                        .value_or { return after_update_error }
-        after_update
+      def update_active_fedora_collection
+        unless params[:update_collection].nil?
+          process_banner_input
+          process_logo_input
+        end
+
+        process_member_changes
+
+        @collection.visibility = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE unless @collection.discoverable?
+        # we don't have to reindex the full graph when updating collection
+        @collection.reindex_extent = Hyrax::Adapters::NestingIndexAdapter::LIMITED_REINDEX
+        if @collection.update(collection_params.except(:members))
+          after_update_response
+        else
+          after_update_errors(@collection.errors)
+        end
+      end
+
+      def update_valkyrie_collection
+        return after_update_errors(form_err_msg(form)) unless form.validate(collection_params)
+
+        result = transactions['change_set.update_collection']
+                 .call(form)
+        @collection = result.value_or { return after_update_errors(result.failure.first) }
+
+        process_member_changes
+
+        unless params[:update_collection].nil?
+          process_banner_input
+          process_logo_input
+        end
```

Comment:
Not sure if this is important, but this PR moves this collection branding processing below the member change processing and update transaction.  I'm not sure if this is an issue or not but thinking that it might change behavior slightly in error cases.  For example, before if the transaction failed then the branding would still have gotten processes but now it wouldn't have.  Actually this sounds like it may be an improvement/bugfix since I would assume there shouldn't be any side effects if the transaction fails.

Reply:
The collection branding is not dependent on the state of the collection object.  So it can go anywhere.  There is another PR (#5458) that fixes some issues with branding and moves the branding processing to transactions.